---
title: "NGB project: data analysis of bacterial and fungal data on *Q. ilex* trees"
author: "Marine Cambon"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output:
  html_document:
    toc: true
    theme: cerulean
    highlight: tango
    code_folding: hide
    number_sections: false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
setwd("~/Documents/Postdoc_Biogeco/1_NGB/exploratory/Qi")
```

# Data loading

The file `"~/Documents/Postdoc_Biogeco/1_NGB/data/ORPHEE_leaf_microbiota/data_Qi_Qr_16S_filtered_formatted.RDS"` and `"~/Documents/Postdoc_Biogeco/1_NGB/data/ORPHEE_leaf_microbiota/data_Qi_Qr_ITS_filtered_formatted.RDS"` are list objects with five elements:

* `samples` are the samples metadata
* `asv_table` is a ASV x samples matrix, containing read counts of ASV per samples
* `assign` is the taxonomic assignment of each ASV
* `asv_sequences` is a fasta file with ASV sequences associated to their name in the other datasets
* `water_potential` is the predawn and midday water potential that have been measured concomitantly with leaf sampling for microbiota analysis

```{r message=FALSE, warning=FALSE}
rm(list=ls())
# Load packages
library(vegan)
library(beanplot)
library(ape)
library(RColorBrewer)
library(knitr)
library(kableExtra)
library(ranger)
source("~/Documents/Postdoc_Biogeco/1_NGB/src/src_function_points_jitter.R")
source("~/Documents/Postdoc_Biogeco/2_DROUGHT/src/src_function_conv_hull_pcoa.R")
paf <- "~/Documents/Postdoc_Biogeco/1_NGB"
  
## Loading 16S data -------------
tmp <- readRDS("~/Documents/Postdoc_Biogeco/1_NGB/data/ORPHEE_leaf_microbiota/data_Qi_Qr_16S_filtered_formatted.RDS")
asv_table_16S <- tmp[["asv_table"]]
samples_16S <- tmp[["samples"]]
assign_16S <- tmp[["assign"]]
wp_16S <- tmp[["water_potential"]]
rm(tmp)
asv_table_clr_16S <- readRDS("/home/mar/Documents/Postdoc_Biogeco/1_NGB/data/ORPHEE_leaf_microbiota/sequencing/processed_16S_sequences/Qi_Qr/aldex.clr_mean_asv.RDS")

# Remove Qr samples from the dataset:
asv_table_16S <- asv_table_16S[,-grep("PU2", colnames(asv_table_16S))]
# Remove ASVs not present in samples:
asv_table_16S <- asv_table_16S[apply(asv_table_16S,1,sum)>0,]
# Remove assignation of ASVs not present in samples:
assign_16S <- assign_16S[match(rownames(asv_table_16S),
                                        rownames(assign_16S)),]
# Remove Qr samples from the metadata
samples_16S <- samples_16S[match(colnames(asv_table_16S),
                                          rownames(samples_16S)),]

# remove endophytes samples from the dataset: 
endo_16S <- subset(samples_16S, sample_type=="endo")
asv_table_endo_16S <- asv_table_16S[,match(rownames(endo_16S),
                                           dimnames(asv_table_16S)[[2]])]

samples_16S <- subset(samples_16S, sample_type=="micro")
asv_table_16S <- asv_table_16S[,match(rownames(samples_16S),
                                      dimnames(asv_table_16S)[[2]])]

## Loading ITS data -------------
tmp <- readRDS("~/Documents/Postdoc_Biogeco/1_NGB/data/ORPHEE_leaf_microbiota/data_Qi_Qr_ITS_filtered_formatted.RDS")
asv_table_ITS <- tmp[["asv_table"]]
samples_ITS <- tmp[["samples"]]
assign_ITS <- tmp[["assign"]]
wp_ITS <- tmp[["water_potential"]]
rm(tmp)
asv_table_clr_ITS <- readRDS("/home/mar/Documents/Postdoc_Biogeco/1_NGB/data/ORPHEE_leaf_microbiota/sequencing/processed_ITS_sequences/Qi_Qr/aldex.clr_mean_asv.RDS")

# Remove Qr samples from the dataset:
asv_table_ITS <- asv_table_ITS[,-grep("PU2", colnames(asv_table_ITS))]
# Remove ASVs not present in samples:
asv_table_ITS <- asv_table_ITS[apply(asv_table_ITS,1,sum)>0,]
# Remove assignation of ASVs not present in samples:
assign_ITS <- assign_ITS[match(rownames(asv_table_ITS),
                                        rownames(assign_ITS)),]
# Remove Qr samples from the metadata
samples_ITS <- samples_ITS[match(colnames(asv_table_ITS),
                                          rownames(samples_ITS)),]

# remove endophytes samples from the dataset: 
endo_ITS <- subset(samples_ITS, sample_type=="endo")
asv_table_endo_ITS <- asv_table_ITS[,match(rownames(endo_ITS),
                                           dimnames(asv_table_ITS)[[2]])]

samples_ITS <- subset(samples_ITS, sample_type=="micro")
asv_table_ITS <- asv_table_ITS[,match(rownames(samples_ITS),
                                      dimnames(asv_table_ITS)[[2]])]

# Colors 
source("~/Documents/Postdoc_Biogeco/1_NGB/src/project_colors_v2.R")
source("~/Documents/Postdoc_Biogeco/1_NGB/src/src_function_points_jitter.R")
source("~/Documents/Postdoc_Biogeco/2_DROUGHT/src/src_function_conv_hull_pcoa.R")
```

Sample metadata are loaded twice because samples that were successfully sequenced and that passed data curation are not exactly the same for 16S and ITS data. For now I don't want to reduce the data sets to their intersection so it is more convenient to keep these duplicated data, but I could think later of a way to merge them.

# Session info

```{r}
sessionInfo()
```